prototype/FFT_image_processing/image_processing.c

```diff
@@ -0,0 +1,414 @@
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <time.h>
+
+#define fft_real float
+
+// project includes
+#include "bitmap.h"
+#include "serial_port.h"
+
+/**
+ * @brief Sends an FFT over serial communication. 
+ * The serial communication must be opened beforehand for the data to be sent.
+ * @param length The length of the FFT array. Must be a power of 2.
+ * @param reals The real part of the FFT array.
+ * This array will contain the end result of the real part of the FFT
+ * @param reals The imaginary part of the FFT array.
+ * This array will contain the end result of the imaginary part of the FFT
+ * @param dir The direction of the FFT. 1 for the FFT, -1 for the inverse FFT.
+ * @return 0 if everything went correctly, 1 otherwise
+*/
+static int FFTOverSerial(uint16_t length, fft_real* reals, fft_real* imgs, int16_t dir) {
+  // Writing
+  // write length
+  int returncode = writeElement(&length, sizeof(length));
+  // write reals
+  if (returncode >= 0) {
+    int writeCode = writeArray(length, reals, sizeof(fft_real));
+    returncode = writeCode < 0 ? writeCode : returncode + writeCode;
+  }
+  // write imaginaries
+  if (returncode >= 0) {
```

Comment:
This variable does not respect our standards

Reply:
I return the number of bytes written. I cannot return error code `1` in case of an error.